src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouppedObject: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const val = String(item[key]);
```

Comment:
What is `val` ?
[This](https://val.ua/) ?
Or [this](https://uk.wikipedia.org/wiki/VAL_(%D0%B3%D1%83%D1%80%D1%82)) ?